### DISCUSSION TITLE
[v8] ColumnDef ergonomics

### DISCUSSION THREAD

**Discussion by Faithfinder** [TanStack/table#4209] [Q&A] [answered]
Docs mostly list the way to define columns as 

```ts
const columns: ColumnDef<TData>[] = [/* definitions */]
```

However, in this case in cell rendering `props.getValue()` is `unknown`, because `TValue` is a second generic argument to `ColumnDef`. And `unknown`  isn't  valid to render in React, type-wise. The only way I see around that is define each column type separately with manually specifying `TValue` type, but ergonomics on that are kinda bad. Need ideas.

**Comment by Faithfinder** [comment, marked as answer]
Found `createColumnHelper` in examples. I wish docs would be fully updated. Oh well.

**Comment by tannerlinsley** [comment]
Working on it!

**Reply by Gusis93** [reply]
So is `createColumnHelper` the only way to infer the type of the accessor? 

If I understand correctly the docs that are right now, if I want to use custom cells and `getValue` is better to use `createColumnHelper.accessor(...)` instead of `accessorKey` ?

**Comment by tannerlinsley** [comment]
Yes

Tanner Linsley

**Reply by Gusis93** [reply]
Thank you
